File: sample/src/main/java/com/pranavpandey/android/dynamic/support/sample/fragment/SettingsFragment.java
/*
 * Copyright 2018 Pranav Pandey
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.pranavpandey.android.dynamic.support.sample.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.AppBarLayout;
import android.support.v4.app.Fragment;

import com.pranavpandey.android.dynamic.support.fragment.DynamicViewPagerFragment;
import com.pranavpandey.android.dynamic.support.sample.R;

import java.util.ArrayList;
import java.util.List;

/**
 * Settings fragment to show app settings and widgets by using
 * {@link DynamicViewPagerFragment}.
 */
public class SettingsFragment extends DynamicViewPagerFragment {

    /**
     * @return The new instance of {@link SettingsFragment}.
     *
     * @param page The default selected page.
     */
    public static Fragment newInstance(int page) {
        SettingsFragment fragment = new SettingsFragment();
        Bundle args = new Bundle();
        args.putInt(ADS_ARGS_VIEW_PAGER_PAGE, page);
        fragment.setArguments(args);

        return fragment;
    }

    @Override
    protected CharSequence getSubtitle() {
        // Set subtitle for the app compat activity.
        return getString(R.string.ads_nav_settings);
    }

    @Override
    protected int setNavigationViewCheckedItem() {
        // Select navigation menu item.
        return R.id.nav_settings;
    }

    @Override
    protected List<String> getTitles() {
        // Initialize an empty string array for tab titles.
        List<String> titles = new ArrayList<>();

        // Add tab titles.
        titles.add(getString(R.string.ads_app));
        titles.add(getString(R.string.ads_widgets));

        // Return all the added tab titles.
        return titles;
    }

    @Override
    protected List<Fragment> getPages() {
        // Initialize an empty fragment array for view pages pages.
        List<Fragment> pages = new ArrayList<>();

        // TODO: Add view pager fragments.
        pages.add(AppSettingsFragment.newInstance());
        pages.add(WidgetsFragment.newInstance());

        // Return all the added fragments.
        return pages;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        // Scroll toolbar for this fragment.
        getDynamicActivity().setToolbarLayoutFlags(
                AppBarLayout.LayoutParams.SCROLL_FLAG_SCROLL
                        | AppBarLayout.LayoutParams.SCROLL_FLAG_ENTER_ALWAYS);

        // Select current page from the bundle arguments.
        if (getArguments() != null && getArguments().containsKey(ADS_ARGS_VIEW_PAGER_PAGE)) {
            setPage(getArguments().getInt(ADS_ARGS_VIEW_PAGER_PAGE));
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();

        // Remove tab layout from the header.
        getDynamicActivity().addHeader(null, true);
    }
}

File: sample/src/main/java/com/pranavpandey/android/dynamic/support/sample/fragment/SettingsFragment.kt
/*
 * Copyright 2018-2021 Pranav Pandey
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.pranavpandey.android.dynamic.support.sample.fragment

import android.os.Bundle
import androidx.fragment.app.Fragment
import com.google.android.material.appbar.AppBarLayout
import com.pranavpandey.android.dynamic.support.fragment.DynamicViewPager2Fragment
import com.pranavpandey.android.dynamic.support.fragment.DynamicViewPagerFragment
import com.pranavpandey.android.dynamic.support.sample.R


/**
 * Settings fragment to show app settings and widgets by using [DynamicViewPagerFragment].
 */
class SettingsFragment : DynamicViewPager2Fragment() {

    companion object {

        /**
         * Returns the new instance of this fragment.
         *
         * @param page The default selected page.
         *
         * @return The new instance of [SettingsFragment].
         */
        fun newInstance(page: Int): androidx.fragment.app.Fragment {
            val fragment = SettingsFragment()
            val args = Bundle()
            args.putInt(DynamicViewPagerFragment.ADS_ARGS_VIEW_PAGER_PAGE, page)
            fragment.arguments = args

            return fragment
        }
    }

    override fun getSubtitle(): CharSequence? {
        // Set subtitle for the app compat activity.
        return getString(R.string.ads_nav_settings)
    }

    override fun getBottomNavigationViewId(): Int {
        // Return the bottom navigation view id.
        return R.id.bottom_navigation
    }

    override fun getCheckedMenuItemId(): Int {
        // Select navigation menu item.
        return R.id.nav_settings
    }

    override fun getTitle(position: Int): String? {
        // TODO: Return tab titles.
        return when (position) {
            1 -> getString(R.string.ads_widgets)
            else -> getString(R.string.ads_app)
        }
    }

    override fun createFragment(position: Int): Fragment {
        // TODO: Return view pager fragments.
        return when (position) {
            1 -> WidgetsFragment.newInstance()
            else -> AppSettingsFragment.newInstance()
        }
    }

    override fun getItemCount(): Int {
        // TODO: Return item count.
        return 2
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        // Scroll toolbar for this fragment.
        dynamicActivity.setToolbarLayoutFlags(
                AppBarLayout.LayoutParams.SCROLL_FLAG_SCROLL
                        or AppBarLayout.LayoutParams.SCROLL_FLAG_ENTER_ALWAYS)
    }
}
